source/common/http/conn_manager_impl.cc

```diff
@@ -404,6 +423,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // Set the trusted address for the connection by taking the last address in XFF.
   downstream_address_ = Utility::getLastAddressFromXFF(*request_headers_);
   decodeHeaders(nullptr, *request_headers_, end_stream);
+
+  Tracing::Decision tracing_decision =
```

Comment:
decodeHeaders() should be the last thing that happens in this function, since it might end up responding. I would move the tracing logic before we call decodeHeaders